math/miller_rabin.cpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2020 @author tjgurwara99
+ * @file
+ *
+ * A basic implementation of Miller-Rabin primality test.
+ */
+
+#include <cassert>
+#include <iostream>
+#include <random>
+#include <vector>
+
+/**
+ * Function to give a binary representation of a number in reverse order
+ * @param num integer number that we want to convert
+ * @return result vector of the number input in reverse binary
+ */
+template <typename T> std::vector<T> reverse_binary(T num) {
+    std::vector<T> result;
+    T temp = num;
+    while (temp > 0) {
+        result.push_back(temp % 2);
+        temp = temp / 2;
+    }
+    return result;
+}
+
+/**
+ * Function for modular exponentiation.
+ * This function is an efficient modular exponentation function.
+ * It can be used with any big integer library such as Boost multiprecision
+ * to give result any modular exponentation problem reletively quickly.
+ * @param base number being raised to a power as integer
+ * @param rev_binary_exponent reverse binary of the power the base is being
+ * raised to
+ * @param mod modulo
+ * @return r the modular exponentiation of $a^{n} \equiv r \mod{m}$ where $n$ is
```

Comment:
in Doxygen, the start and end of inline math functions is: `\f$` and not just `$`.